suite/tests/client-interface/avx512ctx.c

```diff
@@ -0,0 +1,306 @@
+        print("\n");
+    }
+#    endif
+
+    if (memcmp(zmm_buf, zmm_ref, sizeof(zmm_buf)))
+        print("ERROR: wrong zmm value\n");
+    if (memcmp(opmask_buf, opmask_ref, sizeof(opmask_buf)))
+        print("ERROR: wrong mask value\n");
+
+    print("Ok\n");
+}
+
+int
+main()
+{
+    run_avx512();
+    return 0;
+}
+
+#else /* asm code *************************************************************/
+#    include "asm_defines.asm"
+/* clang-format off */
+START_FILE
+
+#ifdef X64
+# define FRAME_PADDING 0
+#else
+# define FRAME_PADDING 0
+#endif
+
+#define FUNCNAME marker
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        sub      REG_XSP, FRAME_PADDING
+        END_PROLOG
+
+        mov      REG_XAX, 0x12345678
+        mov      REG_XAX, 0x12345678
+
+        add      REG_XSP, FRAME_PADDING
+        ret
+        END_FUNC(FUNCNAME)
+#undef FUNCNAME
+
+#define FUNCNAME init_zmm
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        mov      REG_XCX, ARG1
+        PUSH_CALLEE_SAVED_REGS()
+        sub      REG_XSP, FRAME_PADDING
+        END_PROLOG
+
+        vmovups  zmm0,  [REG_XCX]
+        vmovups  zmm1,  [REG_XCX + 0x40]
+        vmovups  zmm2,  [REG_XCX + 0x80]
+        vmovups  zmm3,  [REG_XCX + 0xc0]
+        vmovups  zmm4,  [REG_XCX + 0x100]
+        vmovups  zmm5,  [REG_XCX + 0x140]
+        vmovups  zmm6,  [REG_XCX + 0x180]
+        vmovups  zmm7,  [REG_XCX + 0x1c0]
+#ifdef X64
+        vmovups  zmm8,  [REG_XCX + 0x200]
+        vmovups  zmm9,  [REG_XCX + 0x240]
+        vmovups  zmm10, [REG_XCX + 0x280]
+        vmovups  zmm11, [REG_XCX + 0x2c0]
+        vmovups  zmm12, [REG_XCX + 0x300]
+        vmovups  zmm13, [REG_XCX + 0x340]
+        vmovups  zmm14, [REG_XCX + 0x380]
+        vmovups  zmm15, [REG_XCX + 0x3c0]
+        vmovups  zmm16, [REG_XCX + 0x400]
+        vmovups  zmm17, [REG_XCX + 0x440]
+        vmovups  zmm18, [REG_XCX + 0x480]
+        vmovups  zmm19, [REG_XCX + 0x4c0]
+        vmovups  zmm20, [REG_XCX + 0x500]
+        vmovups  zmm21, [REG_XCX + 0x540]
+        vmovups  zmm22, [REG_XCX + 0x580]
+        vmovups  zmm23, [REG_XCX + 0x5c0]
+        vmovups  zmm24, [REG_XCX + 0x600]
+        vmovups  zmm25, [REG_XCX + 0x640]
+        vmovups  zmm26, [REG_XCX + 0x680]
+        vmovups  zmm27, [REG_XCX + 0x6c0]
+        vmovups  zmm28, [REG_XCX + 0x700]
+        vmovups  zmm29, [REG_XCX + 0x740]
+        vmovups  zmm30, [REG_XCX + 0x780]
+        vmovups  zmm31, [REG_XCX + 0x7c0]
+#endif
+
+        add      REG_XSP, FRAME_PADDING
+        POP_CALLEE_SAVED_REGS()
+        ret
+        END_FUNC(FUNCNAME)
+#undef FUNCNAME
+
+#define FUNCNAME get_zmm
+        DECLARE_FUNC_SEH(FUNCNAME)
+GLOBAL_LABEL(FUNCNAME:)
+        mov      REG_XCX, ARG1
+        PUSH_CALLEE_SAVED_REGS()
+        sub      REG_XSP, FRAME_PADDING
+        END_PROLOG
+
+       vmovups  [REG_XCX],         zmm0
+       vmovups  [REG_XCX + 0x40],  zmm1
+       vmovups  [REG_XCX + 0x80],  zmm2
```

Comment:
suggest: s/0x80/64*2/ which should improve readability

Reply:
Done.